protocols/raft/src/main/java/io/atomix/protocols/raft/impl/RaftServiceManager.java

```diff
@@ -593,51 +603,101 @@ private boolean completeSnapshot(long index) {
   /**
    * Gets or initializes a service context.
    */
-  private DefaultServiceContext getOrInitializeService(ServiceId serviceId, ServiceType serviceType, String serviceName) {
+  private DefaultServiceContext getOrInitializeService(
+      ServiceId serviceId,
+      ServiceType serviceType,
+      String serviceName,
+      int revision,
+      PropagationStrategy propagationStrategy) {
+    if (revision == 0) {
+      DefaultServiceContext service = raft.getServices().getCurrentRevision(serviceName);
+      if (service != null) {
+        return service;
+      } else {
+        revision = 1;
```

Comment:
why 1? will it correct itself after talking with the other services?

Reply:
The default revision if no revision number is specified and no revision exists is 1. If a revision exists then the latest is used.

Reply:
This basically creates a revision of `1` if the service doesn't exist or returns the latest revision if one does exist if `0` (the default) is provided